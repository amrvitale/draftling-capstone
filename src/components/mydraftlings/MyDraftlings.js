import React from 'react';
import { Link } from 'react-router-dom';

class MyDraftlings extends React.Component {
    render() {
        return (
            <div className="draftling1">
                <h1>My Draftlings</h1>
                
                <section class="draftling1">
                    <h2>Draftling Title 1</h2>
                    <p>Sentence preview...</p>
                    <button type="button">Read</button>
                    <button type="button">Edit</button>
                    <button type="button">Delete</button>
                </section>

                <section class="draftling2">
                    <h2>Draftling Title 2</h2>
                    <p>Sentence preview...</p>
                    <button type="button">Read</button>
                    <button type="button">Edit</button>
                    <button type="button">Delete</button>
                </section>

            </div>
        );
    }
}

export default MyDraftlings;