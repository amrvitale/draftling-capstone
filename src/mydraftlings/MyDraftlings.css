* {
    box-sizing: border-box;
  }
  .myDraftlingsPageHero {
    background-image: url(./books.jpg);
    background-repeat: no-repeat; /* Do not repeat the image */
    background-size: cover; /* Resize the background image to cover the entire container */
  }

.container {
    display: flex;
    flex-direction: row;
    flex-flow: row wrap;
    align-items: center;
    justify-content: space-around;


     
  }
.card { 
    margin: 1px;
    padding: 5px;
    border: 1px solid #2B2D42;
    border-radius: 5px;
    background-color: white;
    justify-content: space-evenly;
    

  }

  .container .card {
    padding: 3px;
    display: flex;
    border: 1px #2B2D42;
    
  
    
}

li.card{
    margin: 30px;
    justify-self: center;
    height: 150px;
    width: 320px;
    box-shadow: 3px 3px 5px 6px #ccc;
}

.myDraftlingsHeading {
  -webkit-text-stroke: 1px maroon;
  -webkit-text-fill-color: whitesmoke;    
  font-family: 'Playfair Display', 'Georgia', serif;
    text-shadow:
    3px 3px 0 #000,
    -1px -1px 0 #000,  
     1px -1px 0 #000,
     -1px 1px 0 #000,
      1px 1px 0 #000;
		
}