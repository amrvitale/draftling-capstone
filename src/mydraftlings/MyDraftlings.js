import React from 'react';
import { Link } from 'react-router-dom';
import Draftling from '../draftling/Draftling';
import DraftlingPage from '../draftlingpage/DraftlingPage';
import './MyDraftlings.css'
import background from './bookshelves.jpg'

class MyDraftlings extends React.Component {
    render() {
        console.log(this.props.draftlings);
        return (
          <div className="myDraftlingsPage">
              <div className="myDraftlingsPageHero"> 
                <h1 className="myDraftlingsHeading">My Draftlings Shelf</h1>
                </div>
                        <ul className="container">
                                {this.props.draftlings.map(draftling =>
                                    <li className="card" key={draftling.iddraft}>
                                        <Draftling
                                            iddraft={draftling.iddraft}
                                            title={draftling.title}
                                            wordcount={draftling.wordcount}
                                            genre={draftling.genre}
                                            content={draftling.content}
                                            modified={draftling.modified}
                                        />
                                    </li>
                                )} 
                        </ul>
                     
            </div>
    )};
}
    

export default MyDraftlings;
