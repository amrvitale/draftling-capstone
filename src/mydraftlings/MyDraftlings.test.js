import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MyDraftlings from './MyDraftlings';

describe (`MyDraftlings component`, () => {
    it('renders a .MyDraftlings by default', () => {
        const wrapper = shallow(<MyDraftlings draftlings={[]} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })  
})