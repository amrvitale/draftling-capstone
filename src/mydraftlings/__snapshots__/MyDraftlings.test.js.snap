// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDraftlings component renders a .MyDraftlings by default 1`] = `
<div
  className="myDraftlingsPage"
>
  <div
    className="myDraftlingsPageHero"
  >
    <h1
      className="myDraftlingsHeading"
    >
      My Draftlings Shelf
    </h1>
  </div>
  <ul
    className="container"
  />
</div>
`;
